src/main/java/com/artipie/QuartzScheduler.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie;
+
+import com.artipie.asto.Storage;
+import com.artipie.conda.CondaQuartz;
+import com.artipie.rpm.misc.UncheckedConsumer;
+import com.jcabi.log.Logger;
+import java.util.Collection;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.impl.StdSchedulerFactory;
+
+/**
+ * Quartz scheduler holder: starts scheduler, schedules the jobs and adds hook to shutdown.
+ * @since 0.23
+ */
+public final class QuartzScheduler {
+
+    /**
+     * Abstract storage.
+     */
+    private final Storage asto;
+
+    /**
+     * Repository configurations.
+     */
+    private final Collection<RepoConfig> config;
+
+    /**
+     * Ctor.
+     * @param asto Abstract storage
+     * @param config Repositories
+     */
+    public QuartzScheduler(final Storage asto, final Collection<RepoConfig> config) {
+        this.asto = asto;
+        this.config = config;
+    }
+
+    /**
+     * Start scheduler, add shutdown hook and schedule the jobs.
+     */
+    void setup() {
+        try {
+            final Scheduler scheduler = StdSchedulerFactory.getDefaultScheduler();
+            Runtime.getRuntime().addShutdownHook(
+                new Thread(
+                    () -> {
+                        try {
+                            scheduler.shutdown();
+                            Logger.debug(this, "QuartzScheduler is shut down");
+                        } catch (final SchedulerException err) {
+                            throw new ArtipieException(err);
+                        }
+                    }
+                )
+            );
+            scheduler.start();
+            this.config.stream().filter(repo -> repo.type().equals("conda"))
+                .forEach(
+                    new UncheckedConsumer<>(
+                        conda -> new CondaQuartz(this.asto, conda).triggerJob(scheduler)
+                    )
```

Comment:
Maybe we can provide some common interface from config, e.g. `this.config.stream().flatMap(cfg -> cfg.jobs()).forEach(job -> job.triggerJob(scheduler))`, where `jobs()` can return a collection of jobs to run on quartz scheduler or empty collection if no jobs specified. WDYT?

Reply:
I do not like the idea add new methods about jobs to `RepoConfig` as not each repo needs any jobs. I've added enum with repositories filter and method to schedule the jobs, thus we can add new items there for any repository or even the same job for several repositories.

Reply:
@olenagerasimova I see. But what if need to configure these jobs later? E.g. configure by setting intervals or some additional parameters. Then we need to pass it via repository configuration somehow. How we can pass it to enum values?

Reply:
All additional parameters will be added to `settings` section of the repository configuration `RepoConfig`, an instance of the `RepoConfig` is passed to enum `schedule()` method, where all the job trigger settings can be performed. 

It's as with conda now: cron value is set in the repo config `settings` section and passed over `RepoConfig` instance to enum `schedule()` method and to `CondaQuartz` class.

Reply:
Got it, thanks